codalab/worker/docker_image_manager.py

```diff
@@ -151,14 +148,11 @@ def image_availability_state(image_spec, success_message, failure_message):
                 image = self._docker.images.get(image_spec)
                 digests = image.attrs.get('RepoDigests', [image_spec])
                 digest = digests[0] if len(digests) > 0 else None
-                with self._lock:
-                    self._image_cache[digest] = ImageCacheEntry(
-                        id=image.id,
-                        digest=digest,
-                        last_used=time.time(),
-                        virtual_size=image.attrs['VirtualSize'],
-                        marginal_size=image.attrs['Size'],
-                    )
+                timestamp = str(time.time())
+                for tag in image.tags:
+                    if tag.split(":")[0] == "codalab-image-cache/last-used":
```

Comment:
Create a function that encapsulates the tag logic to avoid duplication with the same thing above.  The string `"codalab-image-cache/last-used"` should ideally only happen once.

Reply:
ah yeah, I was planning to make it a constant but missed it when copy pasting from my old code.